### DISCUSSION TITLE
Per-track fingerprint?

### DISCUSSION THREAD

**Discussion by wusspuss** [paullouisageneau/libdatachannel#985] [Q&A]
the sdp string libdatachannel generates goes as follows:

```
v=0
o=rtc 742103403 0 IN IP4 127.0.0.1
...
a=fingerprint:sha-256 ...
m=video 35837 UDP/TLS/RTP/SAVPF
c=IN IP4 192.168.100.3
...
m=audio 35837 UDP/TLS/RTP/SAVPF
c=IN IP4 192.168.100.3
...
```
other software I'm trying to get it to interact with produces things like
```
v=0
o=- 3719505103060116772 2 IN IP4 127.0.0.1
...
m=video 9 UDP/TLS/RTP/SAVPF 
c=IN IP4 0.0.0.0
a=fingerprint:sha-256 ...
...
m=audio 9 UDP/TLS/RTP/SAVPF
c=IN IP4 0.0.0.0
a=fingerprint:sha-256 ...
```

apparently libdatachannel specifies a fingerprint session-wide, while it can also be specified per media as in the latter example. Is the latter behavior something libdatachannel is compatible with, and how can I get it to do that, particularly with C api?

**Comment by paullouisageneau** [comment]
Fingerprints at media-level should not be an issue. However, if the fingerprints are different, which is weird but technically correct according to RFC 8122, it appears libdatachannel has the same bug as https://github.com/paullouisageneau/libdatachannel/issues/982 for fingerprints, it will incorrectly pick the last one.

**Reply by wusspuss** [reply]
How can I get libdatachannel to produce sdp with fingerprints at media-level?

**Reply by paullouisageneau** [reply]
There is no way to make libdatachannel create SDP with fingerprints at media-level. You can still edit the SDP, but in practice you shouldn't need to do so. What is the other software? Does it support [BUNDLE](https://www.rfc-editor.org/rfc/rfc8843.html)?

**Reply by paullouisageneau** [reply]
Parsing SDP with different fingerprints is fixed by https://github.com/paullouisageneau/libdatachannel/pull/988.
